barretenberg/cpp/src/barretenberg/stdlib/primitives/group/cycle_group.cpp

```diff
@@ -308,28 +332,37 @@ cycle_group<Builder> cycle_group<Builder>::unconditional_add(const cycle_group&
         auto lhs = cycle_group::from_constant_witness(context, get_value());
         // We need to manually propagate the origin tag
         lhs.set_origin_tag(get_origin_tag());
-        return lhs.unconditional_add(other);
+        return lhs.unconditional_add(other, hint);
     }
     if (!lhs_constant && rhs_constant) {
         auto rhs = cycle_group::from_constant_witness(context, other.get_value());
         // We need to manually propagate the origin tag
         rhs.set_origin_tag(other.get_origin_tag());
-        return unconditional_add(rhs);
+        return unconditional_add(rhs, hint);
     }
-
-    const auto p1 = get_value();
-    const auto p2 = other.get_value();
-    AffineElement p3(Element(p1) + Element(p2));
-    if (lhs_constant && rhs_constant) {
-        auto result = cycle_group(p3);
-        // We need to manually propagate the origin tag
-        result.set_origin_tag(OriginTag(get_origin_tag(), other.get_origin_tag()));
-        return result;
+    cycle_group result;
+    if (hint.has_value()) {
+        auto x3 = hint.value().x;
+        auto y3 = hint.value().y;
+        if (lhs_constant && rhs_constant) {
+            return cycle_group(x3, y3, false);
+        }
+        ASSERT(is_point_at_infinity().is_constant() == true);
```

Comment:
Why are we asserting this? I'd expect that if x and y are witnesses, the point at infinity is also going to be one

Reply:
the context of this fn is an unconditional group addition. i.e. we only call this function when we have certainty that a Prover cannot trigger the weierstrass group addition edge cases.

As part of this, the input point cannot be a point at infinity.  `is_point_at_infinity` must both be a circuit constant, and false.

e.g. we use unconditional add when adding together two different generator points. The fixed-base MSM algorithm for linearly independent generator points can also use this method.

I think this assertion should be applied outside of this `if` statement, as well as an assertion that the point at infinity value is false.